sessions/forecast-ensembles.qmd

```diff
@@ -425,467 +474,231 @@ We combine these new ensembles with our previous ensembles in order to make visu
+As we did earlier, we compute one "filtered" ensemble that removes the random walk model.
 
-```{r plot-multiple-weighted-forecasts}
-plot_multiple_weighted_forecasts <- weighted_ensembles |>
-  plot_ensembles(onset_df |> filter(day >= 21)) +
-  lims(y = c(0, 400))
-plot_multiple_weighted_forecasts
+```{r}
+lop_ensemble <- sampled_forecasts_hub |> 
+  hubEnsembles::linear_pool(
+  n_output_samples = 100,
+  model_id = "lop_ensemble",
+  task_id_cols = c("forecast_date", "h", "target_epiweek"), ## columns that define a forecast task
+  compound_taskid_set = c("forecast_date"), ## column that defines a single trajectory
+  derived_task_ids = c("target_epiweek")    ## column that can be derived from the others
+  )   
+
+lop_ensemble_filtered <- sampled_forecasts_hub |> 
+  filter(model_id != "rw") |> 
+  hubEnsembles::linear_pool(
+  n_output_samples = 100,
+  model_id = "lop_ensemble_filtered",
+  task_id_cols = c("forecast_date", "h", "target_epiweek"), ## columns that define a forecast task
+  compound_taskid_set = c("forecast_date"), ## column that defines a single trajectory
+  derived_task_ids = c("target_epiweek")    ## column that can be derived from the others
+  )   
+```
+
+### Visualize sample-based forecasts
+
+To facilitate visual and quantitative comparisons with the earlier ensemble forecasts, we convert the new ensemble forecasts to quantiles and then plot them.
+
+```{r, fig.height = 12}
+all_forecasts <- bind_rows(
+  lop_ensemble,
+  lop_ensemble_filtered
+) |> 
+  hubUtils::convert_output_type(to = list(quantile = quantiles_to_save)) |> 
+  bind_rows(filtered_ensembles)
+
+hubVis::plot_step_ahead_model_output(
+  model_out_tbl = all_forecasts,
+  target_data = flu_data |> 
+    rename(observation = wili) |> 
+    filter(epiweek >= as.Date("2017-09-01")),
+  use_median_as_point = TRUE,
+  x_col_name = "target_epiweek",
+  x_target_col_name = "epiweek",
+  facet = "forecast_date",
+  facet_nrow = 6,
+  intervals = 0.8,
+  fill_transparency = 0.8,
+  interactive = TRUE,
+  pal_color = "Set3"
+)
 ```
 
-and on the log scale.
+::: callout-tip
+### Take 5 minutes
 
-```{r plot-multiple-weighted-forecasts-log}
-plot_multiple_weighted_forecasts +
-  scale_y_log10()
-```
-
-::: {.callout-tip}
-##### Take 2 minutes
-How do the weighted ensembles compare to the simple ensembles?
-Which do you think is best?
-Are you surprised by the results? Can you think of any reasons that would explain them?
+Using the interactive plot, isolate just the `lop_ensemble_filtered` and `median_ensemble_filtered`.
+How and when are they different from each other?
 :::
 
 ::: {.callout-note collapse="true"}
-#### Solution
-How do the weighted ensembles compare to the simple ensembles?
+### Solution
 
+The LOP ensembles tend to be a bit wider than the other ensemble forecasts, especially at times when the forecasts disagree more.
```

Comment:
Is this solution linked to the question? I think both are useful but probably need an edit to align.